### PR TITLE
Issue 944

### PR DESCRIPTION
**Pull request by shroffk**
#944 
Here is the maven release plugin with documentation
@georgweiss @andersharrisson-ess could you have a look at this and let me know if this is good enough to merge.

TODO: test release on branch...
The maven release plugin supports this, but it is basically an action which we could include in the profile but we can also just simply call in via the release command

mvn release:branch -DbranchName=my-branch -DupdateBranchVersions=true

### CODE REVIEW

**Comment by georgweiss** [comment]
Not sure I understand the release profile in phoebus-target/pom.xml. It specifies the release_classpath.py script, but that one only writes some information to a log. Plus there should be a newline char following os.getcwd(). Is the objective to also run mk_classpath.py?

**Comment by shroffk** [comment]
The 'the release_classpath.py' is just a placeholder.

'mk_classpath.py' this is independent from this PR it was there before

**Comment by kasemir** [comment]
I've added 'mk_classpath.py' for those using Eclipse as a helper for creating a .classpath file that lists all dependencies from the target/lib folder. I don't actually use it, not aware of anybody using it, by I keep getting questions about it  ==>  Just remove it.

**Comment by georgweiss** [comment]
Running the build creates dependencies/phoebus-target/release.log. This should be added ti .gitignore

**Comment by shroffk** [comment]
I have added the changes requested by @kasemir and @georgweiss

File: pom.xml [hunk 1]
```diff
@@ -167,10 +170,45 @@
...
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-release-plugin</artifactId>
+            <version>2.5.3</version>
+            <configuration>
+              <preparationGoals>exec:exec</preparationGoals>
+              <checkModificationExcludes>
+                <checkModificationExclude>**/.project</checkModificationExclude>
+                <checkModificationExclude>**/.classpath</checkModificationExclude>
+                <checkModificationExclude>pom.xml</checkModificationExclude>
+                <checkModificationExclude>**/phoebus-target/pom.xml</checkModificationExclude>
```
Comment (andersharrisson-ess) [inline pom.xml]: Out of curiosity, why do you need to exclude these files from the modification check?

**Comment by kasemir** [comment]
>  Generally, I prefer .classpath files kept local

We agreed at the start to support multiple ways of building. As documented in https://github.com/shroffk/phoebus/blob/master/dependencies/README.md and https://github.com/shroffk/phoebus/blob/master/dependencies/README.md you can

a) Depend on the pom.xml for maven to assemble the dependencies from the internet
b) Unzip a target platform and then build with ant
c) ..target platform, import the projects into Eclipse and run

For a) and b), the .classpath files are ignore or overwritten. For c), we use the .classpath files.
If you delete them, we lose option c).

**Comment by andersharrisson-ess** [comment]
> We agreed at the start to support multiple ways of building.

Thank you for the clarification, Kay.
I was not aware of the fact that there were official decisions to support not only Maven and Ant, but also Eclipse. Since that is the case, I have no further objections or comments to this PR.

**Comment by kasemir** [comment]
We can reconsider if there are conflicts, if for example the nbactions.xml added by Claudio or the .gitlab-ci.yml added by you broke the build process for another tool. But so far I think these files can happily coexist.

**Pull request merged**